packages/ERTP/src/mathHelpers/strSetMathHelpers.js

```diff
@@ -62,7 +62,36 @@ const strSetMathHelpers = harden({
     );
     return harden(Array.from(leftSet));
   },
-});
+  doFind: (left, searchParameters) => {
+    const makeIsPrefix = prefix => {
+      const isPrefix = str => str.startsWith(prefix);
+      return isPrefix;
+    };
+    let matchNotFound = false;
+    let arrayOfArrays;
+    try {
+      arrayOfArrays = searchParameters.map(prefix => {
+        const isPrefix = makeIsPrefix(prefix);
+        const arrayOfMatchingStrs = left.filter(isPrefix);
+        if (arrayOfMatchingStrs.length <= 0) {
+          matchNotFound = true;
+          throw Error('match was not found');
+        }
+        return arrayOfMatchingStrs;
```

Comment:
This is anti-frugal. As used by `eventualWithdraw` you're withdrawing the most amount of rights that satisfies the pattern. This is opposite of the nat variant, which leaves me confused about what the abstract meaning of the operation is.